src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ProcessEvent;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ProcessEventRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Status;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.EnvelopeNotFoundException;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event;
+
+import java.util.UUID;
+import javax.transaction.Transactional;
+
+@Service
+public class EnvelopeFinaliserService {
+
+    private final EnvelopeRepository envelopeRepository;
+    private final ProcessEventRepository processEventRepository;
+
+    public EnvelopeFinaliserService(
+        EnvelopeRepository envelopeRepository,
+        ProcessEventRepository processEventRepository
+    ) {
+        this.envelopeRepository = envelopeRepository;
+        this.processEventRepository = processEventRepository;
+    }
+
+    @Transactional
```

Comment:
Nice making it transactional ensures consistency between process events and envelope table 🏅